container/docker/handler.go

```diff
@@ -252,9 +252,11 @@ func newDockerContainerHandler(
 	// split env vars to get metadata map.
 	for _, exposedEnv := range metadataEnvs {
 		for _, envVar := range ctnr.Config.Env {
-			splits := strings.SplitN(envVar, "=", 2)
-			if splits[0] == exposedEnv {
-				handler.envs[strings.ToLower(exposedEnv)] = splits[1]
+			if envVar != "" {
+				splits := strings.SplitN(envVar, "=", 2)
+				if len(splits) == 2 && splits[0] == exposedEnv {
```

Comment:
What is the right behavior when `len(splits) == 1`?

1. Fail silently (current approach)
2. Succeed (`handler.envs[strings.ToLower(exposedEnv)] = ""`)
3. Log a warning

I think 2 or 3 would probably be a better approach, but I'm not sure when this could happne, so present your case. If this "should never happen", we should just log a warning.